vhd/lib/libvhd.c

```diff
@@ -2067,34 +2062,55 @@ vhd_read_block(vhd_context_t *ctx, uint32_t block, char **bufp)
 		return -EINVAL;
 
 	off  = vhd_sectors_to_bytes(blk + ctx->bm_secs);
-	size = vhd_sectors_to_bytes(ctx->spb);
 
 	err  = vhd_footer_offset_at_eof(ctx, &end);
 	if (err)
 		return err;
 
-	err  = posix_memalign(&buf, VHD_SECTOR_SIZE, size);
-	if (err) {
-		err = -err;
-		goto fail;
-	}
-
 	if (end < off + ctx->header.block_size) {
 		size = end - off;
 		memset(buf + size, 0, ctx->header.block_size - size);
 	}
 
+	off = off + vhd_sectors_to_bytes(from);
+
 	err  = vhd_seek(ctx, off, SEEK_SET);
 	if (err)
-		goto fail;
+		return err;
 
 	err  = vhd_read(ctx, buf, size);
+	if (err)
```

Comment:
This is correct, but wouldn't it be simpler to `return vhd_read(ctx, buf, size)`?